core/java/iesi-core/src/main/java/io/metadew/iesi/connection/database/oracle/OracleDatabaseConnectionService.java

```diff
@@ -56,4 +58,13 @@ public Connection getConnection(OracleDatabaseConnection oracleDatabaseConnectio
             throw new RuntimeException(e);
         }
     }
+
+    public String refactorLimitAndOffset(String query) {
+        Pattern pattern = Pattern.compile("(?i)limit(?=\\s+[0-9]+\\s+(?i)offset+\\s+[0-9])");
+        Matcher matcher = pattern.matcher(query);
+        if (matcher.find()) {
+            return query.replaceAll("(?i)limit", "OffSet").replaceAll("(offset|OFFSET|Offset)", "ROWS FETCH NEXT").replaceAll("(ROWS FETCH NEXT\\s+\\w+)", " $1 ROWS ONLY ");
```

Comment:
you can use matcher.start() and matcher.end() to find the beginning and end of the 'limit X offset Y' clause. You can then inject the 'offset Y rows fetch next X rows only' clause using substring(). Note: you've also switched replace values:
offset -> offset rows
limit -> fetch next rows only